pandas/io/tests/test_parsers.py

```diff
@@ -3349,6 +3378,12 @@ def test_whitespace_lines(self):
         df = self.read_csv(StringIO(data))
         tm.assert_almost_equal(df.values, expected)
 
+    def test_read_with_bad_header(self):
+        errmsg = "but only \d+ lines in file"
+        with tm.assertRaisesRegexp(StopIteration, errmsg):
```

Comment:
see what u can do to sync these up and raise s CParserError (for the c parser) or a ValueError for the same op in the python parser - use the oarser as the guide

this maybe pretty tricky as u have to track each change